ts/components/helpers/withErrorModal.tsx

```diff
@@ -62,35 +59,29 @@ const styles = StyleSheet.create({
  *
  * @param WrappedComponent The react component you want to wrap
  * @param errorSelector A redux selector that returns the error (as string) or undefined
- * @param errorMessage The error message to show when errorSelector returns a value
- * @param cancelActionCreator A function that accept the WrappedComponent props and return a redux Action to dispatch.
- *                            We pass in props to make possible to create action like cancelSomething(props.id)
- * @param retryActionCreator A function that accept the WrappedComponent props and return a redux Action to dispatch.
- *                           We pass in props to make possible to create action like loadSomething(props.id)
+ * @param errorMessagesMapping A map that converts an error code from the store (selected by errorSelector) into a
+ *                             human-readable message
+ * @param fallbackError If the error in the store cannot be found in `errorMessagesMapping`, this fallback error is shown
+ * @param onCancel Function that will be called if the user presses the "cancel" button. This will need to clear the
+ *                 stored error to hide the modal
+ * @param onRetry Function that will be called if the user presses the "retry" button. This will need to clear the
+ *                stored error to hide the modal
  */
 export function withErrorModal<P>(
   WrappedComponent: React.ComponentType<P>,
   errorSelector: (state: GlobalState) => Option<string>,
```

Comment:
make it generic (e.g. `type T`, why assume errors are strings?

Reply:
Mostly b/c that's what's stored in the `errors` store but yeah, a generic works better